python/lsst/ap/association/packageAlerts.py

```diff
@@ -405,6 +409,15 @@ def produceAlerts(self, alerts, visit, detector):
                         f.write(alertBytes)
 
         self.producer.flush()
+        visit_midpoint = Time(alert["diaSource"]["midpointMjdTai"], format='mjd', scale='utc')
+        produce_end_timestamp = int(time.time())  # Current time in seconds
+        total_time = produce_end_timestamp - visit_midpoint.unix_tai
+        # Assumed 15 seconds from midpoint and 2 second readout time
+        self.metadata['visit_midpoint'] = visit_midpoint.unix_tai + 15 + 2
```

Comment:
Looking through the rest of the task, you should have access to the science image. You should get the exposure midpoint directly from the image, and you can also get the exposure time so that you can calculate the exposure end point. That won't include the 2s readout time, but we can account for that elsewhere for now.

Reply:
You can get the exposure midpoint in TAI from the science image with:
`mjd_midpoint = exposure.visitInfo.date.toAstropy().tai.mjd`